spec/inputs/http_spec.rb

```diff
@@ -12,6 +13,32 @@
     subject.teardown
   end
 
+  context "with default codec" do
+    subject { LogStash::Inputs::Http.new }
+    context "when receiving a text/plain request" do
+      it "should process the request normally" do
+        subject.register
+        Thread.new { subject.run(queue) }
+        agent.post!("http://localhost:8080/meh.json",
+                    :headers => { "content-type" => "text/plain" },
+                    :body => "hello")
+        event = queue.pop
+        expect(event["message"]).to eq("hello")
+      end
+    end
+    context "when receiving an application/json request" do
+      it "should parse the json body" do
+        subject.register
+        Thread.new { subject.run(queue) }
+        agent.post!("http://localhost:8080/meh.json",
+                    :headers => { "content-type" => "application/json" },
+                    :body => { "message_body" => "Hello" }.to_json)
+        event = queue.pop
+        expect(event["message_body"]).to eq("Hello")
+      end
+    end
+  end
+
```

Comment:
A test with and non existing codec too would be great, I guess it will throw an exception but we could probably give the user a friendlier message?